From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Thu, 3 Jan 2019 13:04:06 -0500
Subject: online-accounts: Set label relationship for listboxes

Set the label relationships so that screen readers
read the headers for the listboxes.

(cherry picked from commit 27b476872ed24372ca3e47b88ffb14abe213481e)
---
 panels/online-accounts/online-accounts.ui | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/panels/online-accounts/online-accounts.ui b/panels/online-accounts/online-accounts.ui
index 6f3ef98..1e70930 100644
--- a/panels/online-accounts/online-accounts.ui
+++ b/panels/online-accounts/online-accounts.ui
@@ -119,7 +119,7 @@
                     <property name="width-request">300</property>
 
                     <child>
-                      <object class="GtkLabel">
+                      <object class="GtkLabel" id="accounts_label">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
                         <property name="label" translatable="yes">Connect to your data in the cloud</property>
@@ -127,6 +127,9 @@
                         <attributes>
                           <attribute name="scale" value="1.66" />
                         </attributes>
+                        <accessibility>
+                          <relation target="accounts_listbox" type="label-for"/>
+                        </accessibility>
                       </object>
                     </child>
                     <child>
@@ -146,12 +149,15 @@
                             <property name="can_focus">True</property>
                             <property name="selection_mode">none</property>
                             <signal name="row-activated" handler="on_listbox_row_activated" object="CcGoaPanel" swapped="yes" />
+                            <accessibility>
+                              <relation target="accounts_label" type="labelled-by"/>
+                            </accessibility>
                           </object>
                         </child>
                       </object>
                     </child>
                     <child>
-                      <object class="GtkLabel">
+                      <object class="GtkLabel" id="providers_label">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
                         <property name="margin-top">20</property>
@@ -160,6 +166,9 @@
                         <attributes>
                           <attribute name="weight" value="bold" />
                         </attributes>
+                        <accessibility>
+                          <relation target="providers_listbox" type="label-for"/>
+                        </accessibility>
                       </object>
                     </child>
                     <child>
@@ -172,6 +181,9 @@
                             <property name="can_focus">True</property>
                             <property name="selection_mode">none</property>
                             <signal name="row-activated" handler="on_provider_row_activated" object="CcGoaPanel" swapped="yes" />
+                            <accessibility>
+                              <relation target="providers_label" type="labelled-by"/>
+                            </accessibility>
                             <child>
                               <object class="GtkListBoxRow" id="more_providers_row">
                                 <property name="visible">True</property>
